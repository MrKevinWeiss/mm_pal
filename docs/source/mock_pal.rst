mock\_pal package
=================

Submodules
----------

mock\_pal.mock\_cli module
--------------------------

.. automodule:: mock_pal.mock_cli
   :members:
   :undoc-members:
   :show-inheritance:

mock\_pal.mock\_dev module
--------------------------

.. automodule:: mock_pal.mock_dev
   :members:
   :undoc-members:
   :show-inheritance:

mock\_pal.mock\_if module
-------------------------

.. automodule:: mock_pal.mock_if
   :members:
   :undoc-members:
   :show-inheritance:


Module contents
---------------

.. automodule:: mock_pal
   :members:
   :undoc-members:
   :show-inheritance:
