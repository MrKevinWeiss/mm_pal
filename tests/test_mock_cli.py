import pytest
from conftest import MM_PATH, EXT_PORT
from mock_pal import MockCli


@pytest.mark.parametrize("kwargs", [{},
                                    {'mm_path': MM_PATH},
                                    {'mem_map': []},
                                    {'driver_type': 'serial'},
                                    {"parser_type": 'json'}])
def test_init(mock_app_json, capsys, kwargs):
    tmp = MockCli(port=EXT_PORT, **kwargs).onecmd("get_version")
    cap = capsys.readouterr()
    assert 'Interface version: 0.0.1' in cap.out


@pytest.mark.parametrize("cmd", ["read_reg i8",
                                 "write_reg i8 1",
                                 "read_struct stt",
                                 "commit",
                                 "soft_reset",
                                 "get_version",
                                 "info_reg arru8",
                                 "info_reg",
                                 "info_param name",
                                 "info_param nothing",
                                 "special_cmd",
                                 "key_error"])
def test_commands(mock_app_json, capsys, regtest, cmd):
    tmp = MockCli(port=EXT_PORT, persistent_history_file="/tmp").onecmd(cmd)
    cap = capsys.readouterr()
    regtest.write(cap.out)


def test_completion(mock_app_json):
    cli = MockCli(port=EXT_PORT)
    assert "arru8" in cli.regs_choices_method()
    assert "name" in cli.param_choices_method()
