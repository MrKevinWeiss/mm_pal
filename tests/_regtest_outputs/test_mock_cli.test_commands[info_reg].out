{'arru16': {'access': 1,
            'array_size': 16,
            'bit_offset': '',
            'bits': '',
            'description': '',
            'name': 'arru16',
            'offset': 512,
            'total_size': 32,
            'type': 'uint16_t',
            'type_size': 2},
 'arru32': {'access': 1,
            'array_size': 256,
            'bit_offset': '',
            'bits': '',
            'description': '',
            'name': 'arru32',
            'offset': 544,
            'total_size': 1024,
            'type': 'uint32_t',
            'type_size': 4},
 'arru8': {'access': 1,
           'array_size': 512,
           'bit_offset': '',
           'bits': '',
           'description': '',
           'name': 'arru8',
           'offset': 0,
           'total_size': 512,
           'type': 'uint8_t',
           'type_size': 1},
 'bf16.b1': {'access': 1,
             'array_size': '',
             'bit_offset': 9,
             'bits': 1,
             'description': '',
             'name': 'bf16.b1',
             'offset': 1583,
             'total_size': '',
             'type': '',
             'type_size': 2},
 'bf16.b6': {'access': 1,
             'array_size': '',
             'bit_offset': 10,
             'bits': 6,
             'description': '',
             'name': 'bf16.b6',
             'offset': 1583,
             'total_size': '',
             'type': '',
             'type_size': 2},
 'bf16.b9': {'access': 1,
             'array_size': '',
             'bit_offset': 0,
             'bits': 9,
             'description': '',
             'name': 'bf16.b9',
             'offset': 1583,
             'total_size': '',
             'type': '',
             'type_size': 2},
 'bf32.b1': {'access': 1,
             'array_size': '',
             'bit_offset': 0,
             'bits': 1,
             'description': '',
             'name': 'bf32.b1',
             'offset': 1585,
             'total_size': '',
             'type': '',
             'type_size': 4},
 'bf32.res': {'access': 1,
              'array_size': '',
              'bit_offset': 1,
              'bits': 31,
              'description': 'Reserved bits',
              'name': 'bf32.res',
              'offset': 1585,
              'total_size': '',
              'type': '',
              'type_size': 4},
 'bf8.b1': {'access': 1,
            'array_size': '',
            'bit_offset': 0,
            'bits': 1,
            'description': '',
            'name': 'bf8.b1',
            'offset': 1582,
            'total_size': '',
            'type': '',
            'type_size': 1},
 'bf8.b2': {'access': 1,
            'array_size': '',
            'bit_offset': 1,
            'bits': 2,
            'description': '',
            'name': 'bf8.b2',
            'offset': 1582,
            'total_size': '',
            'type': '',
            'type_size': 1},
 'bf8.res': {'access': 1,
             'array_size': '',
             'bit_offset': 3,
             'bits': 5,
             'description': 'Reserved bits',
             'name': 'bf8.res',
             'offset': 1582,
             'total_size': '',
             'type': '',
             'type_size': 1},
 'bf8_extra.b1': {'access': 1,
                  'array_size': '',
                  'bit_offset': 0,
                  'bits': 1,
                  'description': '',
                  'name': 'bf8_extra.b1',
                  'offset': 1597,
                  'total_size': '',
                  'type': '',
                  'type_size': 1},
 'bf8_extra.b2': {'access': 1,
                  'array_size': '',
                  'bit_offset': 1,
                  'bits': 2,
                  'description': '',
                  'name': 'bf8_extra.b2',
                  'offset': 1597,
                  'total_size': '',
                  'type': '',
                  'type_size': 1},
 'bf8_extra.res': {'access': 1,
                   'array_size': '',
                   'bit_offset': 3,
                   'bits': 5,
                   'description': 'Reserved bits',
                   'name': 'bf8_extra.res',
                   'offset': 1597,
                   'total_size': '',
                   'type': '',
                   'type_size': 1},
 'i16': {'access': 1,
         'array_size': '',
         'bit_offset': '',
         'bits': '',
         'description': '',
         'name': 'i16',
         'offset': 1576,
         'total_size': '',
         'type': 'int16_t',
         'type_size': 2},
 'i32': {'access': 1,
         'array_size': '',
         'bit_offset': '',
         'bits': '',
         'description': '',
         'name': 'i32',
         'offset': 1578,
         'total_size': '',
         'type': 'int32_t',
         'type_size': 4},
 'i8': {'access': 1,
        'array_size': '',
        'bit_offset': '',
        'bits': '',
        'description': '',
        'name': 'i8',
        'offset': 1575,
        'total_size': '',
        'type': 'int8_t',
        'type_size': 1},
 'stt.arr16': {'access': 1,
               'array_size': 2,
               'bit_offset': '',
               'bits': '',
               'description': '',
               'name': 'stt.arr16',
               'offset': 1590,
               'total_size': 4,
               'type': 'int16_t',
               'type_size': 2},
 'stt.bf16.b1': {'access': 1,
                 'array_size': '',
                 'bit_offset': 9,
                 'bits': 1,
                 'description': '',
                 'name': 'stt.bf16.b1',
                 'offset': 1594,
                 'total_size': '',
                 'type': '',
                 'type_size': 2},
 'stt.bf16.b6': {'access': 1,
                 'array_size': '',
                 'bit_offset': 10,
                 'bits': 6,
                 'description': '',
                 'name': 'stt.bf16.b6',
                 'offset': 1594,
                 'total_size': '',
                 'type': '',
                 'type_size': 2},
 'stt.bf16.b9': {'access': 1,
                 'array_size': '',
                 'bit_offset': 0,
                 'bits': 9,
                 'description': '',
                 'name': 'stt.bf16.b9',
                 'offset': 1594,
                 'total_size': '',
                 'type': '',
                 'type_size': 2},
 'stt.ui8': {'access': 1,
             'array_size': '',
             'bit_offset': '',
             'bits': '',
             'description': '',
             'name': 'stt.ui8',
             'offset': 1589,
             'total_size': '',
             'type': 'uint8_t',
             'type_size': 1},
 'ui16': {'access': 1,
          'array_size': '',
          'bit_offset': '',
          'bits': '',
          'description': '',
          'name': 'ui16',
          'offset': 1569,
          'total_size': '',
          'type': 'uint16_t',
          'type_size': 2},
 'ui32': {'access': 1,
          'array_size': '',
          'bit_offset': '',
          'bits': '',
          'description': '',
          'name': 'ui32',
          'offset': 1571,
          'total_size': '',
          'type': 'uint32_t',
          'type_size': 4},
 'ui8': {'access': 1,
         'array_size': '',
         'bit_offset': '',
         'bits': '',
         'description': '',
         'name': 'ui8',
         'offset': 1568,
         'total_size': '',
         'type': 'uint8_t',
         'type_size': 1},
 'ui8_extra': {'access': 1,
               'array_size': '',
               'bit_offset': '',
               'bits': '',
               'description': '',
               'name': 'ui8_extra',
               'offset': 1596,
               'total_size': '',
               'type': 'uint8_t',
               'type_size': 1}}
